### PR TITLE
Implement video data retrieval & saving

### LINKED ISSUE #32: Implement video data retrieval & saving
### LINKS:
- [ER Diagram](https://drive.google.com/file/d/1ESsI9DQCrAjPEpK6wp-Z0cT65nF0uNyt/view?usp=sharing)

### TO-DO:
- [x] Define DB schema
- [x] Add dependency to manage DB migrations
- [x] Set up DB migrations
- [x] Add Entities representing DB tables
- [x] Set up proper connections between entities in JPA

### TO-DO:
- [x] Create controller
- [x] Implement a service layer that handles data saving

### PR DESCRIPTION
**Pull request by leingenm**
# Pull Request

## Description

Introduced controller and service layer that handles data retrieval and saving into the database.

Videos can be imported by providing a CSV file or a list of video IDs.

Video data retrieval happens in batches and is handled using transactions. If a transaction fails neither video IDs nor video data is saved.

Implemented an endpoint allowing checking of what transactions have failed and which were successful.

## Related Issue

Issue #32 

## Checklist

Please review and check all applicable items:

- [x] I have tested my changes thoroughly
- [x] My code follows the project's coding standards and guidelines
- [x] I have added necessary documentation (if applicable)
- [ ] I have updated relevant tests (if applicable)
- [x] I have reviewed my changes to ensure there are no unnecessary changes
- [x] My pull request has a descriptive title and includes a detailed description of the changes

## Additional Notes

* Updated collections in Postman
* Disabled Liquibase in tests. _Created a separate Spring profile for testing and updated GH Actions step to use it when running tests in CI_

### CODE REVIEW

**Comment by davidrabko** [comment]
image
wow, it's so big

File: build.gradle.kts [hunk 1]
```diff
@@ -54,6 +55,7 @@ dependencies {
     // Test
     testImplementation("org.springframework.boot:spring-boot-starter-test")
     testImplementation("org.springframework.security:spring-security-test")
+    runtimeOnly("com.h2database:h2")
```
Comment (davidrabko) [inline build.gradle.kts]: r u sure we need it here?

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+databaseChangeLog:
+    - include:
+          file: /db/changelog/db.changeset-init-schema.sql
+    - includeAll:
+          path: db/changelog/2024/09
```
Comment (davidrabko) [inline src/main/resources/db/changelog/db.changelog-master.yaml:5]: there are multiple ways to design liquibase projects, primarily two - either `object` oriented or `release` oriented, you can read about those here - https://docs.liquibase.com/start/design-liquibase-project.html

File: build.gradle.kts (hunk 1 above)
Comment (leingenm) [reply build.gradle.kts]: Deleted

**Pull request merged**